resolwe_bio/processes/workflows/wgs_gvcf.py

```diff
@@ -0,0 +1,375 @@
+
+    The pipeline follows GATK best practices recommendations and prepares
+    single-sample paired-end sequencing data for a joint-genotyping step.
+
+    The pipeline steps include Trimmomatic (read trimming), read alignment
+    (BWA-MEM), marking of duplicates (Picard MarkDuplicates), recalibration
+    of base quality scores (ApplyBQSR) and calling of variants
+    (GATK HaplotypeCaller in GVCF mode). The QC reports (FASTQC report,
+    Picard AlignmentSummaryMetrics, CollectWgsMetrics and InsertSizeMetrics)
+    are summarized using MultiQC.
+    """
+
+    slug = "workflow-wgs-gvcf"
+    name = "WGS analysis (GVCF)"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                "image": "public.ecr.aws/s4q6j6e8/resolwebio/dnaseq:6.0.0",
+            },
+        },
+    }
+    data_name = "WGS GVCF analysis ({{ reads|sample_name|default('?') }})"
+    version = "1.0.0"
+    process_type = "data:workflow:wgs:gvcf"
+    category = "Pipeline"
+    entity = {
+        "type": "sample",
+    }
+
+    class Input:
+        """Input fields."""
+
+        reads = DataField("reads:fastq:paired", label="Input sample")
+        ref_seq = DataField("seq:nucleotide", label="Reference sequence")
+        bwa_index = DataField("index:bwa", label="BWA genome index")
+        known_sites = ListField(
+            DataField("variants:vcf"), label="Known sites of variation (VCF)"
+        )
+
+        advanced = BooleanField(
+            label="Show advanced options",
+            description="Inspect and modify parameters.",
+            default=False,
+        )
+
+        class GatkOptions:
+            """Options."""
+
+            intervals = DataField(
+                "bed",
+                label="Use intervals BED file to limit the analysis to the specified parts of the genome.",
+                required=False,
+            )
+
+            contamination = IntegerField(
+                label="Contamination fraction",
+                default=0,
+                description="Fraction of contamination in sequencing data (for all samples) to aggressively remove.",
+            )
+
+        class Trimming:
+            """Trimming parameters."""
+
+            adapters = DataField(
+                "seq:nucleotide",
+                label="Adapter sequences",
+                required=False,
+                description="Adapter sequence in FASTA format that will "
+                "be removed from the read. This field as well as "
+                "Seed mismatches, Simple clip threshold and "
+                "Palindrome clip threshold parameters are  needed to "
+                "perform adapter trimming. Minimum adaptel length and "
+                "Keep both reads are optional parameters.",
+            )
+
+            seed_mismatches = IntegerField(
+                label="Seed mismatches",
+                required=False,
+                disabled="!trimming.adapters",
```

Comment:
Please check if fields are correctly disabled. This field seems to have the wrong group name `trimming` instead of `trimming_options`. Same goes for other fields.

Reply:
Thanks for pointing this out. Disabled fields were not not supported in Resolwe. Fixed in https://github.com/genialis/resolwe/pull/851